pkg/controller/network/vpc/controller.go

```diff
@@ -97,131 +116,127 @@ type external struct {
 }
 
 func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
-	cr, ok := mgd.(*v1alpha3.VPC)
+	cr, ok := mgd.(*v1beta1.VPC)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
 
-	// AWS network resources are uniquely identified by an ID that is returned
-	// on create time; we can't tell whether they exist unless we have recorded
-	// their ID.
 	if meta.GetExternalName(cr) == "" {
-		return managed.ExternalObservation{ResourceExists: false}, nil
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
 	}
 
-	req := e.client.DescribeVpcsRequest(&awsec2.DescribeVpcsInput{
+	response, err := e.client.DescribeVpcsRequest(&awsec2.DescribeVpcsInput{
 		VpcIds: []string{meta.GetExternalName(cr)},
-	})
-	rsp, err := req.Send(ctx)
-	if ec2.IsVPCNotFoundErr(err) {
-		return managed.ExternalObservation{ResourceExists: false}, nil
-	}
+	}).Send(ctx)
+
 	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(err, errDescribe)
+		return managed.ExternalObservation{}, errors.Wrapf(resource.Ignore(ec2.IsVPCNotFoundErr, err), errDescribe)
 	}
 
 	// in a successful response, there should be one and only one object
-	if len(rsp.Vpcs) != 1 {
+	if len(response.Vpcs) != 1 {
 		return managed.ExternalObservation{}, errors.New(errMultipleItems)
 	}
 
-	observed := rsp.Vpcs[0]
+	observed := response.Vpcs[0]
 
 	if observed.State == awsec2.VpcStateAvailable {
 		cr.SetConditions(runtimev1alpha1.Available())
+	} else if observed.State == awsec2.VpcStatePending {
+		cr.SetConditions(runtimev1alpha1.Creating())
 	}
 
-	cr.UpdateExternalStatus(observed)
+	// update the CRD spec for any new values from provider
+	current := cr.Spec.ForProvider.DeepCopy()
+	ec2.LateInitializeVPC(&cr.Spec.ForProvider, &observed)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
```

Comment:
Once you call `e.kube.Update()`, the `cr` object will be populated with whatever the resulting object is in the api-server and that doesn't include `status`, i.e. all changes you make on `status` before this line gets lost. It's probably called only once or twice as late-init frequency is low but would be better if you just moved the `SetConditions` calls below `e.kube.Update`.